connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.planner
+
+import org.scalatest.BeforeAndAfter
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.connect.proto
+import org.apache.spark.sql.catalyst.connect.plans.DslLogicalPlan
+import org.apache.spark.sql.catalyst.connect.plans.DslString
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.test.SharedSparkSession
+
+/**
+ * This suite is based on connect DSL and test that given same dataframe operations, whether
+ * connect could construct a proto plan that can be translated back, and after analyzed, be the
+ * same as Spark dataframe's generated plan.
+ */
+class SparkConnectProtoSuite extends SparkFunSuite
+  with SharedSparkSession with SparkConnectPlanTest with PlanTest with BeforeAndAfter {
+
+  lazy val connectTestRelation =
+    proto.Relation.newBuilder()
+      .setRead(
+        proto.Read.newBuilder()
+        .setNamedTable(proto.Read.NamedTable.newBuilder().addParts("student").build())
+        .build())
+      .build()
+
+  lazy val sparkTestRelation = spark.table("student")
+
+  before {
+    setupTestData()
+  }
+
+  test("Basic select") {
+    val connectPlan = analyze(connectTestRelation.select("id".protoAttr))
+    val sparkPlan = sparkTestRelation.select("id").queryExecution.analyzed
```

Comment:
do we really need DataFrame here? We can use pure logical plans:
```
val testRelation = LocalRelation(...)
...
val connectPlan = transform(...).analyze
val sparkPlan = testRelation.select... // using the catalyst dsl, not DataFrame APIs
```

Reply:
For example, `AnalysisSuite` is a pure logical plan test with dsl.

Reply:
I see what you are saying. In this case we need to support `LocalRelation` in connect proto and let connect plan generated from `LocalRelation` proto (otherwise connect plan will be different from the spark plan which is a LocalRelation).

Is this what you prefer?

Reply:
Yea, it can make the testing simpler. And we need to support it anyway for `SELECT * FROM VALUES ...`

Reply:
Done but with caveats:

1. I added `LocalRelation` without adding the `data` fields. I plan to send follow up PR for that.
2. I added `Attribute` but only have a name field for name.
3. I hardcoded `IntegerType` for all attributes. Current proto does not have data types. I will follow up on this piece.